bolt/source.go

```
@@ -148,6 +163,10 @@ func (c *Client) findSources(ctx context.Context, tx *bolt.Tx, opt platform.Find
 func (c *Client) CreateSource(ctx context.Context, s *platform.Source) error {
 	return c.db.Update(func(tx *bolt.Tx) error {
 		s.ID = c.IDGenerator.ID()
+		// fixme > what if s does not contain a valid OrganizationID ? or contains an empty, thus invaid, OrganizationID ?
+		if !s.OrganizationID.Valid() {
+			s.OrganizationID = c.IDGenerator.ID()
+		}
 
```

@desa I choose to generate a valid `OrganizationID` at creation, since a `0` one would not be valid and it will cause an error (ie., `invalid ID`) during unmarshalling.

---

Do you think it is fine?

---

Or we should throw an error?

---

I think throw an error if it's not valid

---

Throwing an error would make the overall behavior different.

Look at this test case, for example.

https://github.com/influxdata/platform/blob/5e20c766a5fb9038d9501bae9c2c1adfdb19be94/testing/source.go#L60-L64

In the case we choose to throw an error we MUST have and `OrganizationID` for the `platform.Source`.